README.md

```diff
@@ -7388,105 +7387,128 @@ See [SQLCipher 4.0.0 Release](https://www.zetetic.net/blog/2018/11/30/sqlcipher-
+#### Managing the lifetime of passphrase bytes
 
+GRDB offers convenience methods for providing the database passphrases as Swift strings: `usePassphrase(_:)` and `changePassphrase(_:)`. Those methods don't keep the passphrase String in memory longer than necessary. But they are as secure as the standard String type: the lifetime of actual passphrase bytes in memory is not under control.
 
-### Encrypting a Clear-Text Database
+When you want to precisely manage the passphrase bytes, talk directly to SQLCipher, using its raw C functions.
 
-Providing a passphrase won't encrypt a clear-text database that already exists, though. SQLCipher can't do that, and you will get an error instead: `SQLite error 26: file is encrypted or is not a database`.
-
-**To encrypt an existing clear-text database**, create a new and empty encrypted database, and copy the content of the clear-text database in it. The technique to do that is [documented](https://discuss.zetetic.net/t/how-to-encrypt-a-plaintext-sqlite-database-to-use-sqlcipher-and-avoid-file-is-encrypted-or-is-not-a-database-errors/868/1) by SQLCipher. With GRDB, it gives:
+For example:
 
 ```swift
-// The clear-text database
-let clearDBQueue = try DatabaseQueue(path: "/path/to/clear.db")
-
-// The encrypted database, at some distinct location:
-var configuration = Configuration()
-configuration.passphrase = "secret"
-let encryptedDBQueue = try DatabaseQueue(path: "/path/to/encrypted.db", configuration: config)
+import GRDB
+import SQLCipher
 
-try clearDBQueue.inDatabase { db in
-    try db.execute(sql: "ATTACH DATABASE ? AS encrypted KEY ?", arguments: [encryptedDBQueue.path, "secret"])
-    try db.execute(sql: "SELECT sqlcipher_export('encrypted')")
-    try db.execute(sql: "DETACH DATABASE encrypted")
+var config = Configuration()
+config.prepareDatabase = { db in
+    ... // Carefully load passphrase bytes
+    let code = sqlite3_key(db.sqliteConnection, /* passphrase bytes */)
+    ... // Carefully dispose passphrase bytes
+    guard code == SQLITE_OK else {
+        throw DatabaseError(resultCode: code, message: db.lastErrorMessage)
+    }
 }
-
-// Now the copy is done, and the clear-text database can be deleted.
+let dbQueue = try DatabaseQueue(path: dbPath, configuration: config)
 ```
 
-### Attempting Several Passphrases
+#### Passphrase availability vs. Database availability
 
-TODO. Context: synchronizing the passphrase in the keychain, and the passphrase used by the database is tricky: one saving operation may succeed, and the other fail. It is possible that one has to try *two* passphrases:
+When the passphrase is securely stored in the system keychain, your application can protect is using the [`kSecAttrAccessible`](https://developer.apple.com/documentation/security/ksecattraccessible) attribute.
```

Comment:
[...] can protect ~~is~~ **it** using [...]

Reply:
👍